### DISCUSSION TITLE
Wind Speed value for controller: Simulink and dll comparison

### DISCUSSION THREAD

**Discussion by stefanoc1396** [OpenFAST/openfast#1935] [Q&A]
Hello everyone,

I have been testing the controller I have developed in simulink with OpenFAST v3.5, using both simulink with the provided S-Function and by compiling a dll of the controller and launching the simulation using OpenFAST directly.

From the comparison of the results we have noticed that despite the controllers being identical, the results obtained using the two approaches differ. 

In particular we checked the wind speed that is passed to the controller and we noticed that the values seen when using Simulink and the dll compiled version differ.  Additionally, we compared the wind speed passed to the controller with the one reported as output in the .out file, measured at the hub height. As can be observed from the plot, the .out values and the values passed to simulink coincide, but they differ from the values seen when using the dll. 
![image](https://github.com/OpenFAST/openfast/assets/120571255/e24e94d7-f682-4e05-9335-b997bd7c4bee)

When compiling the .dll we passed to the controller the hub wind speed channel, following the interface described here: https://openfast.readthedocs.io/en/dev/source/user/servodyn/ExtendedBladedInterface.html

![image](https://github.com/OpenFAST/openfast/assets/120571255/27e1647e-961b-48fc-85f6-fea6966c3340)

As this difference will result in differences in controller performance, I was wondering if someones has any ideas about the source of this discrepancy, and if further information could be provided about how the wind speed value is passed to the controller when the simulink or bladed interfaces are used.

Thank you in advance for your help!

**Comment by jjonkman** [comment]
Dear @stefanoc1396,

For the DLL interface, are you referring to the `avrSWAP`(27) argument?  This is the horizontal wind speed, which is the vector magnitude of the U and V components.

Which wind speed output are you comparing this to?  My guess is you are comparing this wind speed to a specific wind speed component, e.g. `Wind1VelX` = U.

Best regards,
